pathfinder.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"context"
+	"runtime"
+	"sync"
+
+	"golang.org/x/mod/module"
+
+	"github.com/CrowdStrike/perseus/perseusapi"
+)
+
+// newPathFinder initializes and returns a new [pathFinder] instance using the provided Perseus
+// client, maximum depth, and status callback.
+func newPathFinder(c perseusapi.PerseusServiceClient, maxDepth int, status func(string)) pathFinder {
+	return pathFinder{
+		c:        c,
+		maxDepth: maxDepth,
+		status:   status,
+	}
+}
+
+// pathFinder queries the Perseus database to contruct dependency paths of up to maxDepth steps between
+// two modules.
+type pathFinder struct {
+	c        perseusapi.PerseusServiceClient
+	status   func(string)
+	maxDepth int
+
+	sem chan struct{}
+	wg  *sync.WaitGroup
```

Comment:
nit: any reason this is a pointer?

Reply:
Habit to avoid copying the wait group (and having the linter yell at me) more than anything.  Looking at the code we never copy a `pathFinder` instance so this could be a value instead.